tests/zfs-tests/cmd/libzfs_input_check.c

```diff
@@ -549,7 +549,6 @@ test_recv_new(const char *dataset, int fd)
 	fnvlist_add_string(props, "org.openzfs:launch", "September 17th, 2013");
 	fnvlist_add_nvlist(optional, "localprops", props);
 	fnvlist_add_boolean(optional, "force");
-	fnvlist_add_boolean(optional, "heal");
```

Comment:
This looks like a typo.  "heal" was added as part of the corrective receive PR, #9372.

Reply:
I mentioned this in a comment; the libzfs_input test was failing. The issue is that `heal` only works with an existing dataset. Previously, we would always fail with ZFS_ERR_STREAM_TRUNCATED because we always pulled the next record in `dmu_recv_begin` before the begin checks happened. With this change, we only pull the next record in after the begin checks, so if we have `heal` set, the receive fails with ENOENT because the snapshot doesn't exist. If we make the snapshot exist instead, we fail with EEXIST.  It's possible there's a better way to fix the test; perhaps we can run twice, once with `heal` not in the required list (and expect the stream truncated error) and once with it in the required list (and expect ENOENT) ?

Reply:
Ah, I see thanks for the explanation.  I missed the comment since I was looking at the squashed version of the patches.  I think what we'd probably want to do, as a separate change, is add new receive test which correctly uses `heal` with an existing dataset.  But that doesn't need to hold this up.